### PR TITLE
feat(oas): declare x-expanded-relations - Admin

### PR DESCRIPTION
**Pull request by patrick-medusajs**
## What

Declare `x-expanded-relations` on admin responses

## Why

Allows our codegen to alter generated types to better represent the shape of models included in responses when the backend augments the models with relations and/or calculated totals.

## How

* Consider defaultRelations in route bootstrapping. Include as `relations`.
* Consider relations passes to services in controllers. Move to a defaultRelations const. Include as `relations`.
* Consider relations in service sub-processes. Include as `implicit`.
* Consider eager relations in TypeORM models. Include as `eager`.
* Consider decoratedTotals fields. Include as `totals`

Extra scope:
* Fix responses that are not respecting OAS response contract.

## Tests

* Run `yarn install`
* Run `yarn build`
* Open generated types in `packages/generated/client-types/src/lib/models`
* Expect types decorate with x-expanded-relations to be mutated using `SetRelation`. 
* Expect `Merge` for be used for deeply nested relations.

### CODE REVIEW

File: packages/medusa/src/api/routes/admin/returns/index.ts [hunk 1]
```diff
@@ -43,6 +101,11 @@ export type AdminReturnsCancelRes = {
 /**
  * @schema AdminReturnsListRes
  * type: object
+ * x-expanded-relation:
+ *   field: returns
+ *   relations:
+ *     - order
+ *     - swap
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/returns/index.ts:108]: https://github.com/medusajs/medusa/blob/40473a253b328097c672db7a482d09f692ecff2a/packages/medusa/src/api/routes/admin/returns/index.ts#L36-L36

File: packages/medusa/src/api/routes/admin/returns/index.ts [hunk 2]
```diff
@@ -70,6 +133,10 @@ export type AdminReturnsListRes = PaginatedResponse & {
 /**
  * @schema AdminReturnsRes
  * type: object
+ * x-expanded-relation:
+ *   field: return
+ *   relations:
+ *     - swap
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/returns/index.ts:139]: https://github.com/medusajs/medusa/blob/40473a253b328097c672db7a482d09f692ecff2a/packages/medusa/src/api/routes/admin/returns/index.ts#L35

File: packages/medusa/src/api/routes/admin/shipping-options/get-shipping-option.ts [hunk 3]
```diff
@@ -52,7 +54,11 @@
 export default async (req, res) => {
   const { option_id } = req.params
   const optionService = req.scope.resolve("shippingOptionService")
-  const data = await optionService.retrieve(option_id)
+
+  const data = await optionService.retrieve(option_id, {
+    select: defaultFields,
+    relations: defaultRelations,
+  })
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/shipping-options/get-shipping-option.ts:61]: Harmonize response with create-shipping-option
https://github.com/medusajs/medusa/blob/e3bf93406484cc871a78d178276afe05c7195fa5/packages/medusa/src/api/routes/admin/shipping-options/create-shipping-option.ts#L106-L111

File: packages/medusa/src/api/routes/admin/shipping-options/index.ts [hunk 4]
```diff
@@ -55,6 +55,15 @@ export const defaultRelations = ["region", "profile", "requirements"]
 /**
  * @schema AdminShippingOptionsListRes
  * type: object
+ * x-expanded-relations:
+ *   field: shipping_options
+ *   relations:
+ *     - profile
+ *     - region
+ *     - requirements
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/shipping-options/index.ts:63]: https://github.com/medusajs/medusa/blob/e3bf93406484cc871a78d178276afe05c7195fa5/packages/medusa/src/api/routes/admin/shipping-options/index.ts#L53-L53

File: packages/medusa/src/api/routes/admin/shipping-options/index.ts [hunk 5]
```diff
@@ -82,6 +91,15 @@ export type AdminShippingOptionsListRes = PaginatedResponse & {
 /**
  * @schema AdminShippingOptionsRes
  * type: object
+ * x-expanded-relations:
+ *   field: shipping_option
+ *   relations:
+ *     - profile
+ *     - region
+ *     - requirements
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/shipping-options/index.ts:99]: https://github.com/medusajs/medusa/blob/e3bf93406484cc871a78d178276afe05c7195fa5/packages/medusa/src/api/routes/admin/shipping-options/index.ts#L53-L53

File: packages/medusa/src/api/routes/admin/shipping-profiles/index.ts [hunk 6]
```diff
@@ -69,6 +69,11 @@ export type AdminDeleteShippingProfileRes = DeleteResponse
 /**
  * @schema AdminShippingProfilesRes
  * type: object
+ * x-expanded-relations:
+ *   field: shipping_profile
+ *   relations:
+ *     - products
+ *     - shipping_options
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/shipping-profiles/index.ts:76]: https://github.com/medusajs/medusa/blob/e3bf93406484cc871a78d178276afe05c7195fa5/packages/medusa/src/api/routes/admin/shipping-profiles/index.ts#L44-L45

File: packages/medusa/src/api/routes/admin/shipping-profiles/update-shipping-profile.ts [hunk 7]
```diff
@@ -91,7 +95,11 @@ export default async (req, res) => {
       .update(profile_id, validated)
   })
 
-  const data = await profileService.retrieve(profile_id)
+  const data = await profileService.retrieve(profile_id, {
+    select: defaultAdminShippingProfilesFields,
+    relations: defaultAdminShippingProfilesRelations,
+  })
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/shipping-profiles/update-shipping-profile.ts:101]: Harmonize response with get-shipping-profile
https://github.com/medusajs/medusa/blob/e3bf93406484cc871a78d178276afe05c7195fa5/packages/medusa/src/api/routes/admin/shipping-profiles/get-shipping-profile.ts#L65-L68

File: packages/medusa/src/api/routes/admin/swaps/index.ts [hunk 8]
```diff
@@ -22,18 +22,18 @@ export default (app) => {
...
   "fulfillments",
+  "order",
   "payment",
+  "return_order",
   "shipping_address",
   "shipping_methods",
-  "cart",
-  "cart.items",
-  "cart.items.variant",
-  "cart.items.adjustments",
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/swaps/index.ts:36]: no change, simply sorted lines

File: packages/medusa/src/api/routes/admin/variants/index.ts [hunk 9]
```diff
@@ -75,6 +75,12 @@ export const defaultAdminVariantFields: (keyof ProductVariant)[] = [
 /**
  * @schema AdminVariantsListRes
  * type: object
+ * x-expanded-relations:
+ *   field: variants
+ *   relations:
+ *     - options
+ *     - prices
+ *     - product
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/variants/index.ts:83]: https://github.com/medusajs/medusa/blob/e3bf93406484cc871a78d178276afe05c7195fa5/packages/medusa/src/api/routes/admin/variants/index.ts#L48-L48

File: packages/medusa/src/api/routes/admin/variants/index.ts [hunk 10]
```diff
@@ -102,6 +108,12 @@ export type AdminVariantsListRes = PaginatedResponse & {
 /**
  * @schema AdminVariantsRes
  * type: object
+ * x-expanded-relations:
+ *   field: variant
+ *   relations:
+ *     - options
+ *     - prices
+ *     - product
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/variants/index.ts:116]: https://github.com/medusajs/medusa/blob/e3bf93406484cc871a78d178276afe05c7195fa5/packages/medusa/src/api/routes/admin/variants/index.ts#L48-L48

**Review by patrick-medusajs — COMMENTED** [review]
self-review: highlighting noteworthy code changes

**Review by olivermrbl — APPROVED** [review]
LGTM!

**Comment by patrick-medusajs** [comment]
We can see the result of the these changes on the generated types on this commit (https://github.com/medusajs/medusa/commit/0bee5645c4458554ee25fed6e868a5e519a06509) from a future PR.

**Pull request merged**